README.md

```diff
@@ -55,29 +55,60 @@ The criteria for picking these event names included:
-   - `status`: result of the test. Can `passed`, `failed` or `skipped`. A skipped test is disabled, i.e. it will not be executed.
-   - `runtime`: execution time in milliseconds
-   - `errors`: array containing all errors. Depending on the framework, this is a single exception or a list of failed assertions. Will be empty for statuses other than failed.
- - **Suite**: A suite is a collection of tests and potentially other suites.
-   - `name`: name of the suite
-   - `childSuites`: array with all direct subsuites
-   - `tests`: array containing all tests that directly belong to the suite (but not to a child suite)
-   - `runtime`: execution time of the whole suite in milliseconds (including child suites)
-   - `status`: summarized status of the suite
-      - `failed`, if at least one test failed
-      - `skipped`, if all tests in the suite are skipped (and there is at least one skipped test)
-      - `passed`, if there is at least one passed test and all other tests are skipped or if there are no tests in the suite.
+Based on the discussion in [#12](https://github.com/js-reporters/js-reporters/issues/12#issuecomment-120483356), there are two basic data structures: Suites and Tests. A test represents an atomic test/spec/`it()`. A suite contains tests and optionally other suites.
+
+For `testStart` and `testEnd`, the corresponding test object is passed to the reporter. The same applies to `suiteStart` and `suiteEnd` where the matching suite object is passed to the reporter. For `runStart` and `runEnd` a "global" suite object is passed to the reporter, this suite wraps all other top-level user defined suites as its child suites, it will be reffered to as `globalSuite`.
+
+The data structures are defined as follows:
+
+- **Suite**: `Object` - A suite is a collection of tests and potentially other suites.
+ - **name**: `String|undefined` - name of the suite, will be `undefined` only for the `globalSuite`
+ - **fullname**: `Array` - array of strings containing the name of the suite and the names of all its suites ancestors
+ - **tests**: `Array` - array containing all tests that directly belong to the suite (but not to a child suite)
+ - **childSuites**: `Array` - array with all direct subsuites
+ - **status**: `String|undefined` - summarized status of the suite, it will be `undefined` on the `suiteStart` and `runtStart` event
+    - `failed`, if at least one test in the suite or in its child suites has failed
+    - `skipped`, if all tests in the suite and in its child suites are skipped (and there is at least one skipped test)
+    - `passed`, if there is at least one passed test in the suite or in its child suites and all other tests are skipped or if there are no tests in the suite.
+ - **testCounts**: `Object` - contains how many tests have passed, failed etc. including the tests of child suites
+    - **passed**: `Number|undefined` - number of passed tests, it will be `undefined` for the `runStart` or `suiteStart` event
```

Comment:
Should be "will be undefined for the A **and** B event" (not "or"), this isn't boolean logic. Applies to several other instances as well.